Android compatibility

satheshshiva opened the pull request:
Provides android compatibility of this library. Out of the box support which runs on Android API 28 (Android Pie)

Commit:
test connection

Commit:
refactored test folder

Commit:
added the missed out java classes

Commit:
replaced org.apache.httpcomponents with cz.msebera.android.httpclient

Commit:
using maven install plugin

Commit:
using maven install plugin

Commit:
shaded plugin inclusion

Commit:
read me update

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/signed)](https://cla.opensource.microsoft.com/OfficeDev/ews-java-api?pullRequest=690) <br/>All CLA requirements met.

Commit:
update tests for auto discover

Commit:
enable tracing

Commit:
Tests/Examples for Push subscription and streaming mail notification

Commit:
made the id and watermark as public

Commit:
Attachments support:
1) Made the FileAttachment.java as public so that an attachment id can stored and the object can be retrieved again to download the attachment at a later time without loading the email.

2) Added the setter method for the attachment id for the same above reason.

Commit:
made a small change in the test

Commit:
adding contact read test
